libraries/resource_ssl_certificate_readers.rb

```diff
@@ -85,10 +85,11 @@ def data_bag_read_fail(desc, db, type = 'data bag')
             "#{db[:bag]}.#{db[:item]}[#{db[:key]}]"
         end
 
-        def read_from_data_bag(bag, item, key, encrypt = false, secret = nil)
+        def read_from_data_bag(bag, item, key, encrypt = false, secret_file = nil)
           unsafe_no_exceptions_block do
             data =
               if encrypt
+                secret_file.nil? ? secret = nil : secret = Chef::EncryptedDataBagItem.load_secret(secret_file)
```

Comment:
I think this can be simplified to the following:

```ruby
secret = secret_file.nil? ? nil : Chef::EncryptedDataBagItem.load_secret(secret_file)
```